### PR TITLE
Makefile: modified to run tests serially

### LINKED ISSUE #162: Run tests serially in Makefile
Instead of using `go test ./... -v` in the `tests` target of the Makefile, get a list of all the go packages and range over them, testing one at a time.  Also remove the `godotenv -f /etc/alpine-hodler/auth.env` line, this is deprecated functionality.

Removing godotenv will also require an update to CONTRIBUTING.md.

### PR DESCRIPTION
**Pull request by snapdgn**
This closes #162 
@prestonvasquez would be grateful if you could please take a look :), please do suggest changes, if any.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -1,6 +1,14 @@
+PKGS     = $(or $(PKG),$(shell env GO111MODULE=on $(GO) list ./...))
+TESTPKGS = $(shell env GO111MODULE=on $(GO) list -f \
+            '{{ if or .TestGoFiles .XTestGoFiles }}{{ .ImportPath }}{{ end }}' \
+            $(PKGS))
+GO 	= go
```
Comment (prestonvasquez) [inline Makefile:5]: Great idea! I have three comments

1. Rename `GO` to `GO_COMPILER`, `GC`, `GO_BIN`, or anything more descriptive
2. Add a comment, something like `# GC is the go compiler/binary for compiling Go data.`
3. Remove the white space before the `=`.

**Review by prestonvasquez — CHANGES_REQUESTED** [review]
Awesome job! I just have a few comments :)

File: Makefile (hunk 1 above)
Comment (snapdgn) [reply Makefile:5]: will do :)

File: Makefile [hunk 2]
```diff
@@ -15,13 +23,13 @@ proto:
 # test runs all of the application tests locally.
 .PHONY: tests
 tests:
-	go clean -testcache
-	godotenv -f /etc/alpine-hodler/auth.env go test ./... -v
```
Comment (prestonvasquez) [inline Makefile:19]: Could we also remove these lines from the CI scrip: https://github.com/alpine-hodler/gidari/blob/main/scripts/run-ci-tests.sh#L5-L6

**Comment by prestonvasquez** [comment]
I think we can also delete this file: https://github.com/alpine-hodler/gidari/blob/main/scripts/list-pkgs.sh

Your solution for fetching the package list is outstanding!

File: Makefile [hunk 3]
```diff
@@ -1,6 +1,17 @@
+# PKGS returns all Go packages in the Gidari code base.
+PKGS     = $(or $(PKG),$(shell env GO111MODULE=on $(GC) list ./...))
+# TESTPKGS returns all Go packages int the Gidari code base that contain "*_test.go" files.
```
Comment (prestonvasquez) [inline Makefile:3]:
```suggestion

# TESTPKGS returns all Go packages int the Gidari code base that contain "*_test.go" files.
```

File: Makefile [hunk 4]
```diff
@@ -1,6 +1,17 @@
+# PKGS returns all Go packages in the Gidari code base.
+PKGS     = $(or $(PKG),$(shell env GO111MODULE=on $(GC) list ./...))
```
Comment (prestonvasquez) [inline Makefile:2]:
```suggestion
PKGS = $(or $(PKG),$(shell env GO111MODULE=on $(GC) list ./...))
```

File: Makefile [hunk 5]
```diff
@@ -1,6 +1,17 @@
...
+# TESTPKGS returns all Go packages int the Gidari code base that contain "*_test.go" files.
+TESTPKGS = $(shell env GO111MODULE=on $(GC) list -f \
+            '{{ if or .TestGoFiles .XTestGoFiles }}{{ .ImportPath }}{{ end }}' \
+            $(PKGS))
+# GC is the go compiler/binary for compiling Go data.
```
Comment (prestonvasquez) [inline Makefile:7]:
```suggestion

# GC is the go compiler/binary for compiling Go data.
```

**Review by prestonvasquez — APPROVED** [review]
Looking great, just a few small cleanup items and I'll get this merged!

**Comment by prestonvasquez** [comment]
Thank you @snapdgn ! Please feel free to pick up more issues, add issues, or join our [discord server](https://discord.com/invite/3jGYQz74s7) if you feel like discussing the project further!

**Comment by snapdgn** [comment]
sure, will do.Thanks a lot :)

**Pull request merged**